File: bench/list.go
package main
import "fmt"
type L struct { h int; t *L }
func build(n int, acc *L) *L { for ; n > 0; n-- { acc = &L{n, acc} }; return acc }
func sum(xs *L) int { s := 0; for ; xs != nil; xs = xs.t { s += xs.h }; return s }
func main() { tot := 0; for k := 0; k < 30; k++ { tot += sum(build(1000000, nil)) }; fmt.Println(tot) }

File: bench/list.rs
struct L { h: i64, t: Option<Box<L>> }
fn main() { let mut tot = 0i64; for _ in 0..30 { let mut acc: Option<Box<L>> = None; for n in (1..=1000000i64).rev() { acc = Some(Box::new(L { h: n, t: acc })); }
  let mut s = 0i64; let mut p = &acc; while let Some(c) = p { s += c.h; p = &c.t; } tot += s;
  while let Some(mut c) = acc { acc = c.t.take(); } } println!("{}", tot); }
